jormungandr/src/jrpc/eth_transaction/logic.rs

```diff
@@ -137,12 +145,38 @@ pub async fn estimate_gas(tx: Transaction, context: &Context) -> Result<Number,
         .into())
 }
 
-pub fn sign(_address: H160, _message: Bytes, _context: &Context) -> Result<H512, Error> {
-    // TODO implement
-    Ok(H512::zero())
+pub fn sign(address: H160, message: Bytes, context: &Context) -> Result<H512, Error> {
+    let account_secret = context
+        .try_full()?
+        .evm_keys
+        .iter()
+        .find(|&sec| sec.address() == address)
+        .ok_or(Error::AccountSignatureError)?;
+    let signed = eip_191_signature(message, account_secret)?;
+    let (recovery_id, sig_bytes) = signed.serialize_compact();
+    let signature = {
+        let mut sig = [0u8; 65];
+        sig[..64].copy_from_slice(&sig_bytes[..]);
+        sig[64] = (recovery_id.to_i32() % 2) as u8;
+        sig
+    };
+    Ok(H512::from_slice(&signature[..]))
 }
 
-pub fn call(_tx: Transaction, _number: BlockNumber, _context: &Context) -> Result<Bytes, Error> {
-    // TODO implement
-    Ok(Default::default())
+pub async fn call(tx: Transaction, _: BlockNumber, context: &Context) -> Result<Bytes, Error> {
+    let account_secret = context
```

Comment:
You can remove all these parts related to signing and replace it with the following
```
    let blockchain_tip = context.blockchain_tip()?.get_ref().await;
     Ok(blockchain_tip
         .ledger()
         .call_evm_transaction(tx.into())
         .map_err(Box::new)?
         .into())
```

Reply:
Done. Thank you.